### DISCUSSION TITLE
Setting res for model run

### DISCUSSION THREAD

**Discussion by mullerrobert799** [Deltares/Wflow.jl#288] [Q&A] [answered]
Thanks for your support on my other questions!!

I observed the default res in HydroMT-Wflow is: res=0.008333333333333333. What is the basis for it? Can we increase the resolution to 0.0225 or more? If we increase it, then what would be the effect on results (discharge)? Would you guide me the criteria of choosing the resolution of model?

Second question is that: if area of model is greater than 45,000 sq.km, then it is not possible to run HydroMT_wflow even with 1TB RAM then how could we get the HydroMT_wflow successful? any guidance on it to handle the large basins? 

Reference link:
[WflowModel.setup_basemaps(region, res=0.008333333333333333, hydrography_fn='merit_hydro', basin_index_fn='merit_hydro_index', upscale_method='ihu')[[source]](https://deltares.github.io/hydromt_wflow/latest/_modules/hydromt_wflow/wflow.html#WflowModel.setup_basemaps)[#](https://deltares.github.io/hydromt_wflow/latest/_generated/hydromt_wflow.WflowModel.setup_basemaps.html#hydromt_wflow.WflowModel.setup_basemaps)](https://deltares.github.io/hydromt_wflow/latest/_generated/hydromt_wflow.WflowModel.setup_basemaps.html)

**Comment by hboisgon** [comment]
Second question answered here: https://github.com/Deltares/hydromt/issues/330

**Comment by verseve** [comment, marked as answer]
> I observed the default res in HydroMT-Wflow is: res=0.008333333333333333. What is the basis for it? Can we increase the resolution to 0.0225 or more? If we increase it, then what would be the effect on results (discharge)? Would you guide me the criteria of choosing the resolution of model?

This is indeed the default model resolution. The assumption is that performance of wflow_sbm at this model resolution is also for smaller (headwater) catchments optimal and this is mainly based on an application for the [Rhine basin](https://agupubs.onlinelibrary.wiley.com/doi/full/10.1029/2019WR026807), which showed that discharge deteriorates for headwater catchments on coarser resolutions. A [large sample assessment](https://hess.copernicus.org/articles/26/4407/2022/hess-26-4407-2022.html) of small- to medium-sized basins (CAMELS dataset) showed no optimal grid size resolution (3 km, 1 km, and 200 m) throughout the domain, although individual catchments could show large differences in performance.

Generally, we recommend a maximum grid size of ~5 km. It also depends of course on your modelling objective, for simulating discharge at a (large) catchment outlet a coarser resolution than ~1 km is most probably fine, but for small catchments and for example snow dynamics this may not be the case as a result of terrain smoothing and/or locality of forcing variables.

**Comment by mullerrobert799** [comment]
Thanks a lot for your guidance and it helped a lot, as we couldn't see any difference between 1km res and 5km res results (discharge). It means there is no need to run model at a finer scale until we don't have any specific objective.

In case catchment is very big and it is not possible to simulate it with even 5km res due to memory issues, then what options do we have? can we split the catchment for wflow simulation? any suggestion? many thanks for your support!

**Comment by JoostBuitink** [comment]
Regarding the memory issues, I think @hboisgon gave a detailed answer (in this issue https://github.com/Deltares/hydromt/issues/330) on how to deal with this. We have successfully build and run numerous models for catchments with areas larger than 150.000 km2, on machines with 16GB of RAM. Running Wflow for these models only requires a couple of GB's RAM (also when using 1km grid size). 

If memory does become an issue, that it is possible to split a catchment into smaller sections, and link the outflow of an upstream model to the downstream model via the `inflow` parameter (which becomes a forcing parameter at that stage). In your case, a catchment with an area of 45.000 km2 should be no problem to run with Wflow on most laptops.
